src/elements/List/ListItem.js

```diff
@@ -50,11 +93,34 @@ ListItem.propTypes = {
   /** Additional classes. */
   className: PropTypes.string,
 
+  /** Shorthand for primary content. */
+  content: customPropTypes.contentShorthand,
+
+  /** Shorthand for ListDescription. */
+  description: customPropTypes.itemShorthand,
+
   /** A list item can disabled. */
   disabled: PropTypes.bool,
 
+  /** Shorthand for ListHeader. */
+  header: customPropTypes.itemShorthand,
+
+  /** Shorthand for ListIcon. */
+  icon: customPropTypes.every([
```

Comment:
I think that's more of an exception, tbh. A quick search shows it's only applicable in the header and list while icon is a prop in over 20 component/subcomponents (although maybe we're not applying the `disallow` in some places we should be?).

In any case, the `*Shorthand` props includes a `disallow(['children'])` because that's required in *all* cases. This is only applicable in some cases so I think it'll make it more clear to apply the `disallow` within the component's prop definitions that need it.